### PR TITLE
Initial codebase

### PR DESCRIPTION
**Pull request by mats-claassen**

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[flake8]
+max-line-length = 130
+per-file-ignores =
+    __init__.py: F401
+exclude =
+    .git,__pycache__,
```
Comment (pgrill) [inline .flake8:6]: Can we add the end of line?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+WORKDIR /app
+COPY ./requirements.txt /app/
+
+RUN pip install -r requirements.txt
+RUN apt-get -y update
+RUN apt-get -y install git
+RUN apt-get install ffmpeg libsm6 libxext6  -y
+
+COPY . /app/
+WORKDIR /app/
```
Comment (pgrill) [inline Dockerfile:12]: Is this `WORKDIR` line required?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+Clone this repository and run:
+```
+docker build -t spoter_embeddings .
+docker run --rm -it --entrypoint=bash --gpus=all -v $PWD:/app spoter_embeddings
+```
+
+> Running without specifying the `entrypoint` will train the model with the hyperparameters specified in `train.sh`
+
+If you prefer running in a **virtual environment** instead, then first install dependencies:
```
Comment (pgrill) [inline README.md:56]: We should specify the python version required

File: datasets/__init__.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# flake8: noqa
```
Comment (pgrill) [inline datasets/__init__.py:1]: Taking into account that you included the 
```
per-file-ignores =
    __init__.py: F401
```
rule in the `.flake8` file the `noqa` sentence is not required

File: datasets/czech_slr_dataset.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        if self.transform:
+            depth_map = self.transform(depth_map)
+
+        return depth_map, label
+
+    def __len__(self):
+        return len(self.labels)
+
+
+if __name__ == "__main__":
```
Comment (pgrill) [inline datasets/czech_slr_dataset.py:75]: Is this line required?

File: datasets/datasets_utils.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+import numpy as np
+from torch.nn.utils.rnn import pad_sequence
+from random import randrange
+
+from augmentations import augment_arm_joint_rotate, augment_rotate, augment_shear
+from normalization.body_normalization import BODY_IDENTIFIERS
+from normalization.hand_normalization import HAND_IDENTIFIERS
+
+
+HAND_IDENTIFIERS = [id + "_0" for id in HAND_IDENTIFIERS] + [id + "_1" for id in HAND_IDENTIFIERS]
```
Comment (pgrill) [inline datasets/datasets_utils.py:14]: This line is duplicated in the file [augmentations/augment.py](https://github.com/xmartlabs/spoter-embeddings/pull/1/files#diff-371de4fe12589d89e9720c55dd8feae20cd7abf0c219a9e68b33a06aed66f389).

Maybe we can have it defined in a single place. What do you think?

File: datasets/datasets_utils.py [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+
+def load_dataset(file_location: str):
+
+    # Load the datset csv file
+    df = pd.read_csv(file_location, encoding="utf-8")
+
+    # TO BE DELETED
```
Comment (pgrill) [inline datasets/datasets_utils.py:22]: Should we delete the line? Otherwise we should delete de comment

File: models/spoter_model.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        self.transformer.decoder.layers = _get_clones(custom_decoder_layer, self.transformer.decoder.num_layers)
+
+    def forward(self, inputs):
+        h = torch.unsqueeze(inputs.flatten(start_dim=1), 1).float()
+        h = self.transformer(self.pos + h, self.class_query.unsqueeze(0)).transpose(0, 1)
+        res = self.linear_class(h)
+
+        return res
+
+
```
Comment (pgrill) [inline models/spoter_model.py:68]: Is this line required?

File: requirements.txt [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+notebook==6.5.2
+opencv-python==4.6.0.66
+pandas==1.1.5
+pandas==1.1.5
+plotly==5.11.0
+scikit-learn==1.0.2
+torchvision==0.13.0
+tqdm==4.54.1
+# requests==2.28.1
+# onnx==1.12.0
```
Comment (pgrill) [inline requirements.txt:16]: Can we remove the commented lines?

File: Dockerfile (hunk 2 above)
Comment (mats-claassen) [reply Dockerfile:12]: No

File: datasets/__init__.py (hunk 4 above)
Comment (mats-claassen) [reply datasets/__init__.py:1]: Good catch

File: datasets/czech_slr_dataset.py (hunk 5 above)
Comment (mats-claassen) [reply datasets/czech_slr_dataset.py:75]: This was copied from the Spoter repo, that is why it was left as it was but I can remove it

File: datasets/datasets_utils.py (hunk 7 above)
Comment (mats-claassen) [reply datasets/datasets_utils.py:22]: I can review that and fix

File: requirements.txt (hunk 9 above)
Comment (mats-claassen) [reply requirements.txt:16]: There is a Readme in the `web` folder that explains how to convert a model and run it in a web browser. For that we need these commented dependencies. What I could do to make this clearer is:
* Document this in the main readme
* create a separate requirements.txt file with all the requirements for the model conversion use case

Comment (pgrill) [reply requirements.txt]: Make sense for me

**Pull request merged**